mainapp/views.py

```diff
@@ -84,3 +107,24 @@ def profile(request):
     else:
         form = ProfileForm()
         return render(request, 'mainapp/profile.html', {'form': form})
+
+
+def done(request):
+    prof_id = Profile.objects.get(user=request.user).id
+    prod = Product.objects.all().filter(author_id=prof_id, is_active=False)
```

Comment:
Can't be reduced with next sample:

```python3
prod = Product.objects.filter(author__user=request.user, is_active=False).all()
```

`Queryset.all` must be called after filtering for avoid repeat query to db.

Reply:
understood and fixed